### PR TITLE
Add raw_status

### PR DESCRIPTION
**Pull request by rtwalker**
My attempt at #1942. 
 
With respect to the tests, I tried to replicate what was done with `TerminationStatus`, but I'm unsure whether I did so successfully.

### CODE REVIEW

File: docs/src/solutions.md [hunk 1]
```diff
@@ -52,6 +52,12 @@ This function will return a `MOI.TerminationStatusCode` `enum`.
 MOI.TerminationStatusCode
 ```
 
+Additionally, we can receive a solver specific string explaning why the
+optimization stopped with
+```julia
+raw_status(model)
+```
```
Comment (odow) [inline docs/src/solutions.md]: It's better to go 
```
optimization stopped with [`raw_status`](@ref).
```

and then add `raw_status` to the reference list at the bottom.

**Comment by blegat** [comment]
> Could you also test calling RawStatusString when it hasn't been set so that it returns nothing?

It would rather have it either return a String or throw OptimizeNotCalled. It should not return `nothing`

**Comment by blegat** [comment]
The issue with the failing tests is due to https://github.com/JuliaOpt/MathOptInterface.jl/issues/707 which will be fixed in https://github.com/JuliaOpt/MathOptInterface.jl/pull/726.
We can wait for MOI v0.9 to be released.

**Comment by blegat** [comment]
@rtwalker You should be able to finish it now as MOI v0.9 has been released and JuMP master has been update to MOI v0.9

**Comment by rtwalker** [comment]
Hi, I just opened this to look at what needed finishing up and saw that this was merged. I guess everything was ok?

**Comment by blegat** [comment]
Yes, I just relaunch CI now that MOI v0.9 was released and it worked :)

**Pull request merged**